pkg/controller/util/endpoint/controller_utils.go

```diff
@@ -164,27 +164,39 @@ func PodChanged(oldPod, newPod *v1.Pod, endpointChanged EndpointsMatch) (bool, b
 		labelsChanged = true
 	}
 
-	// If the pod's deletion timestamp is set, remove endpoint from ready address.
-	if newPod.DeletionTimestamp != oldPod.DeletionTimestamp {
+	// If the pod's deletion timestamp has been set, remove endpoint from ready address.
+	if oldPod.DeletionTimestamp == nil && newPod.DeletionTimestamp != nil {
 		return true, labelsChanged
 	}
+
 	// If the pod's readiness has changed, the associated endpoint address
 	// will move from the unready endpoints set to the ready endpoints.
 	// So for the purposes of an endpoint, a readiness change on a pod
 	// means we have a changed pod.
 	if podutil.IsPodReady(oldPod) != podutil.IsPodReady(newPod) {
 		return true, labelsChanged
 	}
-	// Convert the pod to an Endpoint, clear inert fields,
-	// and see if they are the same.
-	// TODO: Add a watcher for node changes separate from this
-	// We don't want to trigger multiple syncs at a pod level when a node changes
-	return endpointChanged(newPod, oldPod), labelsChanged
```

Comment:
This was comparing the whole v1.EndpointAddress object but the `TargetRef.ResourceVersion` field, is it safe to compare only the IPs now?